pandas/tests/indexing/test_loc.py

```diff
@@ -1645,6 +1645,21 @@ def test_loc_setitem_mask_td64_series_value(self):
         assert expected == result
         tm.assert_frame_equal(df, df_copy)
 
+    @pytest.mark.parametrize("indexing_func", [list, np.array])
+    @pytest.mark.parametrize("func", ["loc", "iloc"])
+    @pytest.mark.parametrize("rhs_func", [list, np.array])
+    def test_loc_setitem_boolean_list(self, func, rhs_func, indexing_func):
+        # GH#20438 testing specifically list key, not arraylike
+        ser = Series([0, 1, 2])
+        getattr(ser, func)[indexing_func([True, False, True])] = rhs_func([5, 10])
```

Comment:
in the np.array case, do we care if the ndarray has bool vs object dtype?

Reply:
Object dtype is killed by https://github.com/pandas-dev/pandas/blob/b96665777c083516c9cdf21d25b7d9ccf70c81bd/pandas/core/indexers.py#L153:L157 and raises a ValueError too.